nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/util/TlsHelper.java

```diff
@@ -0,0 +1,177 @@
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequest;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequestBuilder;
+import org.bouncycastle.util.io.pem.PemWriter;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.security.auth.x500.X500Principal;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.SignatureException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+
+public class TlsHelper {
+    public static final String PROVIDER = BouncyCastleProvider.PROVIDER_NAME;
+    public static final String PKCS12 = "PKCS12";
+    private final KeyPairGenerator keyPairGenerator;
+    private final int days;
+    private final String signingAlgorithm;
+
+    public TlsHelper(TlsHelperConfig tlsHelperConfig) throws NoSuchAlgorithmException {
+        this(tlsHelperConfig.getDays(), tlsHelperConfig.getKeySize(), tlsHelperConfig.getKeyPairAlgorithm(), tlsHelperConfig.getSigningAlgorithm());
+    }
+
+    public TlsHelper(TlsToolkitCommandLine tlsToolkitCommandLine) throws NoSuchAlgorithmException {
+        this(tlsToolkitCommandLine.getTlsHelperConfig());
+    }
+
+    public TlsHelper(int days, int keySize, String keyPairAlgorithm, String signingAlgorithm) throws NoSuchAlgorithmException {
+        this(createKeyPairGenerator(keyPairAlgorithm, keySize), days, signingAlgorithm);
+    }
+
+    protected TlsHelper(KeyPairGenerator keyPairGenerator, int days, String signingAlgorithm) {
+        this.keyPairGenerator = keyPairGenerator;
+        this.days = days;
+        this.signingAlgorithm = signingAlgorithm;
+    }
+
+    public static void addBouncyCastleProvider() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private static KeyPairGenerator createKeyPairGenerator(String algorithm, int keySize) throws NoSuchAlgorithmException {
+        KeyPairGenerator instance = KeyPairGenerator.getInstance(algorithm);
+        instance.initialize(keySize);
+        return instance;
+    }
+
+    public KeyPair generateKeyPair() throws NoSuchAlgorithmException {
+        return keyPairGenerator.generateKeyPair();
+    }
+
+    public void addToKeyStore(KeyStore keyStore, KeyPair keyPair, String alias, char[] passphrase, Certificate... certificates) throws GeneralSecurityException, IOException {
```

Comment:
I think all these methods need some kind of documentation, ideally public Javadoc, especially as this method could be interpreted by a caller to add a public key certificate to a truststore and would expose the private key instead.